nibabel/streamlines/tck.py

```diff
@@ -411,37 +407,36 @@ def _read(cls, fileobj, header, buffer_size=4):
                 if eof:
                     buff = buff[:n_read]
 
-                buffs.append(buff)
-
-                # Make sure we've read enough to find a streamline delimiter.
-                if fiber_marker not in buff:
-                    # If we've read the whole file, then fail.
-                    if eof:
-                        # Could have minimal buffering, and have read only the
-                        # EOF delimiter
-                        buffs = [bytearray().join(buffs)]
-                        if not buffs[0] == eof_marker:
-                            raise DataError(
-                                "Cannot find a streamline delimiter. This file"
-                                " might be corrupted.")
-                    else:
-                        # Otherwise read a bit more.
-                        continue
-
-                all_parts = bytearray().join(buffs).split(fiber_marker)
-                point_parts, buffs = all_parts[:-1], all_parts[-1:]
-                point_parts = [p for p in point_parts if p != b'']
-
-                for point_part in point_parts:
-                    # Read floats.
-                    pts = np.frombuffer(point_part, dtype=dtype)
-                    # Convert data to little-endian if needed.
-                    yield pts.astype('<f4', copy=False).reshape([-1, 3])
-
-                n_streams += len(point_parts)
-
-            if not buffs[-1] == eof_marker:
-                raise DataError("Expecting end-of-file marker 'inf inf inf'")
+                raw_values = np.frombuffer(buff, dtype=dtype)
+
+                # Convert raw_values into a list of little-endian triples (for x,y,z coord)
+                coords = raw_values.astype('<f4', copy=False).reshape((-1, 3))
+
+                # Find stream delimiter locations (all NaNs)
+                delims = np.where(np.isnan(coords).all(axis=1))[0]
+
+                # Recover leftovers, which can't have delimiters in them
+                if leftover.size:
+                    delims += leftover.shape[0]
+                    coords = np.vstack((leftover, coords))
+
+                begin = 0
+                for delim in delims:
+                    pts = coords[begin:delim]
+                    if pts.size:
+                        yield coords[begin:delim]
+                        n_streams += 1
+                    begin = delim + 1
+
+                # The rest gets appended to the leftover
```

Comment:
```suggestion
                # The rest becomes the new leftover.
```

Reply:
Could you apply this suggestion, as well?

Reply:
done!